### PR TITLE
Alternate reduce allocations

### PR DESCRIPTION
**Pull request by JonRowe**
An alternative to #936, this takes the low hanging fruit but leaves the more controversial diff change in behaviour behind.

### CODE REVIEW

**Comment by mrageh** [comment]
How does this reduce the number of objects allocated?
The reasoning in #936 was hard to follow...

**Comment by JonRowe** [comment]
> How does this reduce the number of objects allocated?
The reasoning in #936 was hard to follow...

The `lamba` keyword creates a new instance each time, where as assigning it to a constant does not.

**Comment by myronmarston** [comment]
LGTM

**Comment by mrageh** [comment]
Thanks @JonRowe, :ship: it
